Clinical-Genomics/scout
# Feature multiple loqusdb 200612

This PR adds a functionality or fixes a bug.

**How to test**:
Test for different scenarios.

1. Configure in Scout GUI
Update loqusdbID in Scout. See that loqusdbID is updated

2. Configure in Command Line
Run command line update of loqusdbID. See that loqusdbID is updated

3. Run Configured LoqusDB
* Configure  loqusdb with id
* Configure a loqusdb-id for testing institute.
-Reviewing a variant will use configured liquids
4. Fallback to Default
* Configure a 'default' loqusdb.
* Do *not* configure a loqusdb-id for testing institute.
-Reviewing a variant will use *default* LoqusDB

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**



Configure LoqusDB per Institute
===============================

It is now possible to configure an individual LoqusDB per institute. This feature is 
backwards compatible. 

### Highlights
* Backwards compatible
* If multiple loqusdb:s are configured in LOQUSDB_SETTINGS, one *should* have id 'default'
* Institutes missing 'LoqusDB id' configuration will use 'default'
* If 'LoqusDB id' configured for an Institute is not found 
  `ConfigError: LoqusDB id not found` is raised


To use only one LoqusDB, edit 'config.py'. 

    LOQUSDB_SETTINGS = dict(
    binary_path=</path/to/bin/loqusdb>
    config_path=<path/to/loqus/config>    # Optional
    }


Setup Multiple LoqusDB
----------------------

1. Edit config.py
To configure a separate LoqusDB give it reference "id". If needed configure an (optional)
config_path. A default id should always exist.

```
    LOQUSDB_SETTINGS = [
        {"binary_path": </path/to/bin/loqusdb>, "id": "default", "config_path":"loqus.cfg"},
        {"binary_path": </path/to/bin/loqusdb>, "id": "foo", "config_path":"bar.cfg"}
    ]
```

2.1 Configure Institute via Scout Browser
Go to sidebar 'Settings'. Find 'LoqusDB id' and enter the configured id from 
config.py. Click Save. Your configuration is now active.

![Screenshot 2020-07-10 at 12 52 16](https://user-images.githubusercontent.com/1150065/87147271-9ea50600-c2ac-11ea-9f66-333b37783d52.png)

URL for institute settings: `http://localhost:5000/overview/<institute>/settings`


2.2 Configure Institute via Command Line

    $ scout update institute  <institute> --loqusdb_id <loqusdb_id>


3. Configure LoqusDB
See [LoqusDB page](https://github.com/moonso/loqusdb)
